src/backend/routers/agent.py

```diff
@@ -1,3 +1,144 @@
-import factory
+from fastapi import APIRouter, Depends, Form, HTTPException, Request
 
-from backend.database_models.agent import Agent
+from backend.config.routers import RouterName
+from backend.crud import agent as agent_crud
+from backend.database_models.agent import Agent as AgentModel
+from backend.database_models.database import DBSessionDep
+from backend.schemas.agent import Agent, CreateAgent, DeleteAgent, UpdateAgent
+from backend.services.auth.utils import get_header_user_id
+
+router = APIRouter(
+    prefix="/v1/agents",
+)
+router.name = RouterName.AGENT
+
+
+@router.post("", response_model=Agent)
+def create_agent(session: DBSessionDep, agent: CreateAgent, request: Request):
+    user_id = get_header_user_id(request)
```

Comment:
Looks like we don't use user_id anywhere, so you actually don't need to call `get_header_user_id`

Reply:
Same thing for all endpoints

Reply:
Yup that works!